NXP FRDM K66F example for micro speech does not even build

System information

    OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04
    TensorFlow installed from (source or binary): source
    TensorFlow version: bfc8733
    GCC/Compiler version (if compiling from source): arm-none-eabi-g++ 8.2.1

The example I'm running can be found here: 

https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/micro/examples/micro_speech#deploy-to-nxp-frdm-k66f

Describe the problem
When running

$ make -f tensorflow/lite/micro/tools/make/Makefile TARGET=mbed TAGS="nxp_k66f" generate_micro_speech_mbed_project

it results in the following error:

/make/Makefile TARGET=mbed TAGS="nxp_k66f" generate_micro_speech_mbed_project
tensorflow/lite/micro/tools/make/Makefile:313: warning: overriding recipe for target 'tensorflow/lite/micro/tools/make/downloads/person_model_int8'
tensorflow/lite/micro/tools/make/Makefile:313: warning: ignoring old recipe for target 'tensorflow/lite/micro/tools/make/downloads/person_model_int8'
make: *** No rule to make target 'tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed/third_party/gemmlowp/fixedpoint/fixedpoint.h', needed by 'generate_micro_speech_mbed_project'. Stop.

**Update**
I tried again on my other machine:
OS: Ubuntu 20.04.1 LTS x86_64
Kernel: 5.4.0-47-generic
TensorFlow:  b36436b
This time the following step(3) worked:
` make -f tensorflow/lite/micro/tools/make/Makefile TARGET=mbed TAGS="nxp_k66f" generate_micro_speech_mbed_project`
However, at step 7, when trying to compile I get the following error:
**Output **
[mbed] Working path "/home/carlo/Work/Demos/Speech/3rd/tensorflow/tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed" (library)
[mbed] Program path "/home/carlo/Work/Demos/Speech/3rd/tensorflow/tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed"
[mbed] WARNING: Missing Python modules were not auto-installed.
       The Mbed OS tools in this program require the following Python modules: click
       You can install all missing modules by running "pip install -r requirements.txt" in "/home/carlo/Work/Demos/Speech/3rd/tensorflow/tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed/mbed-os"
       On Posix systems (Linux, etc) you might have to switch to superuser account or use "sudo"
---
/home/carlo/Work/Demos/Speech/3rd/tensorflow/tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed/mbed-os/tools/config/__init__.py:1154: SyntaxWarning: "is" with a literal. Did you mean "=="?
  (unit_kind is "application" or
Building project mbed (K66F, GCC_ARM)
Scan: mbed
Compile [  0.4%]: mbed_tz_context.c
Compile [  0.7%]: CAN.cpp
Compile [  1.1%]: DigitalIn.cpp
Compile [  1.5%]: I2CSlave.cpp
Compile [  1.9%]: DigitalInOut.cpp
Compile [  2.2%]: Ethernet.cpp
Compile [  2.6%]: AnalogOut.cpp
Compile [  3.0%]: BusIn.cpp
Compile [  3.3%]: DigitalOut.cpp
Compile [  3.7%]: BusOut.cpp
Compile [  4.1%]: InterruptManager.cpp
[Warning] InterruptManager.h@204,40: 'CallChain' is deprecated: CallChain has been deprecated and will be removed. [-Wdeprecated-declarations]
[Warning] InterruptManager.cpp@63,60: 'CallChain' is deprecated: CallChain has been deprecated and will be removed. [-Wdeprecated-declarations]
[Warning] InterruptManager.cpp@92,32: 'CallChain' is deprecated: CallChain has been deprecated and will be removed. [-Wdeprecated-declarations]
Compile [  4.5%]: BusInOut.cpp
Compile [  4.8%]: InterruptIn.cpp
Compile [  5.2%]: ResetReason.cpp
Compile [  5.6%]: PortIn.cpp
Compile [  5.9%]: TableCRC.cpp
Compile [  6.3%]: PortInOut.cpp
Compile [  6.7%]: QSPI.cpp
Compile [  7.1%]: PortOut.cpp
Compile [  7.4%]: PwmOut.cpp
Compile [  7.8%]: SerialWireOutput.cpp
Compile [  8.2%]: AnalogIn.cpp
[Error] mstd_type_traits@191,12: 'std::conditional_t' has not been declared
[Error] mstd_type_traits@193,12: 'std::enable_if_t' has not been declared
[Error] mstd_type_traits@210,51: expected template-name before '<' token
[Error] mstd_type_traits@210,51: expected '{' before '<' token
[Error] mstd_type_traits@217,51: expected template-name before '<' token
[Error] mstd_type_traits@217,51: expected '{' before '<' token
[Error] mstd_type_traits@1084,12: 'std::remove_const_t' has not been declared
[Error] mstd_type_traits@1086,12: 'std::remove_volatile_t' has not been declared
[Error] mstd_type_traits@1088,12: 'std::remove_cv_t' has not been declared
[Error] mstd_type_traits@1090,12: 'std::add_const_t' has not been declared
[Error] mstd_type_traits@1092,12: 'std::add_volatile_t' has not been declared
[Error] mstd_type_traits@1094,12: 'std::add_cv_t' has not been declared
[Error] mstd_type_traits@1096,12: 'std::remove_reference_t' has not been declared
[Error] mstd_type_traits@1173,12: 'std::make_signed_t' has not been declared
[Error] mstd_type_traits@1175,12: 'std::make_unsigned_t' has not been declared
[Error] mstd_type_traits@1177,12: 'std::remove_extent_t' has not been declared
[Error] mstd_type_traits@1179,12: 'std::remove_all_extents_t' has not been declared
[Error] mstd_type_traits@1181,12: 'std::remove_pointer_t' has not been declared
[Error] mstd_type_traits@1183,12: 'std::add_pointer_t' has not been declared
[Error] mstd_type_traits@1185,12: 'std::aligned_storage_t' has not been declared
[Error] mstd_type_traits@1187,12: 'std::decay_t' has not been declared
[Error] mstd_type_traits@1189,12: 'std::common_type_t' has not been declared
[Error] mstd_type_traits@1191,12: 'std::result_of_t' has not been declared
[Error] mstd_type_traits@1195,42: 'remove_cv_t' is not a member of 'std'; did you mean 'remove_cv'?
[Error] mstd_type_traits@1195,42: 'remove_cv_t' is not a member of 'std'; did you mean 'remove_cv'?
[Error] mstd_type_traits@1195,59: 'remove_reference_t' is not a member of 'std'; did you mean 'remove_reference'?
[Error] mstd_type_traits@1195,78: template argument 1 is invalid
[Error] mstd_type_traits@1195,79: expected '{' before '>' token
[Error] mstd_type_traits@1240,10: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
[Error] mstd_type_traits@1240,21: expected initializer before '<' token
[Error] mstd_type_traits@1250,10: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
[Error] mstd_type_traits@1250,21: expected initializer before '<' token
[Error] mstd_type_traits@1260,10: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
[Error] mstd_type_traits@1260,21: expected initializer before '<' token
[Error] mstd_type_traits@1271,10: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
[Error] mstd_type_traits@1271,21: expected initializer before '<' token
[Error] mstd_type_traits@1281,10: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
[Error] mstd_type_traits@1281,21: expected initializer before '<' token
[Error] mstd_type_traits@1291,10: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
[Error] mstd_type_traits@1291,21: expected initializer before '<' token
[Error] mstd_type_traits@1302,10: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
[Error] mstd_type_traits@1302,21: expected initializer before '<' token
[ERROR] In file included from <command-line>:
././BUILD/K66F/GCC_ARM-RELEASE/mbed_config.h:66: warning: "NDEBUG" redefined
   66 | #define NDEBUG                                                                            // defined by application
      | 
<command-line>: note: this is the location of the previous definition
In file included from ./mbed-os/platform/mbed_atomic.h:883,
                 from ./mbed-os/platform/SingletonPtr.h:24,
                 from ./mbed-os/drivers/AnalogIn.h:25,
                 from ./mbed-os/drivers/source/AnalogIn.cpp:18:
./mbed-os/platform/cxxsupport/mstd_type_traits:191:12: error: 'std::conditional_t' has not been declared
  191 | using std::conditional_t;
      |            ^~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:193:12: error: 'std::enable_if_t' has not been declared
  193 | using std::enable_if_t;
      |            ^~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:210:51: error: expected template-name before '<' token
  210 | struct conjunction<B1, BN...> : std::conditional_t<bool(B1::value), conjunction<BN...>, B1> { };
      |                                                   ^
./mbed-os/platform/cxxsupport/mstd_type_traits:210:51: error: expected '{' before '<' token
./mbed-os/platform/cxxsupport/mstd_type_traits:217:51: error: expected template-name before '<' token
  217 | struct disjunction<B1, BN...> : std::conditional_t<bool(B1::value), B1, disjunction<BN...>> { };
      |                                                   ^
./mbed-os/platform/cxxsupport/mstd_type_traits:217:51: error: expected '{' before '<' token
In file included from ./mbed-os/platform/mbed_atomic.h:883,
                 from ./mbed-os/platform/SingletonPtr.h:24,
                 from ./mbed-os/drivers/AnalogIn.h:25,
                 from ./mbed-os/drivers/source/AnalogIn.cpp:18:
./mbed-os/platform/cxxsupport/mstd_type_traits:1084:12: error: 'std::remove_const_t' has not been declared
 1084 | using std::remove_const_t;
      |            ^~~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1086:12: error: 'std::remove_volatile_t' has not been declared
 1086 | using std::remove_volatile_t;
      |            ^~~~~~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1088:12: error: 'std::remove_cv_t' has not been declared
 1088 | using std::remove_cv_t;
      |            ^~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1090:12: error: 'std::add_const_t' has not been declared
 1090 | using std::add_const_t;
      |            ^~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1092:12: error: 'std::add_volatile_t' has not been declared
 1092 | using std::add_volatile_t;
      |            ^~~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1094:12: error: 'std::add_cv_t' has not been declared
 1094 | using std::add_cv_t;
      |            ^~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1096:12: error: 'std::remove_reference_t' has not been declared
 1096 | using std::remove_reference_t;
      |            ^~~~~~~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1173:12: error: 'std::make_signed_t' has not been declared
 1173 | using std::make_signed_t;
      |            ^~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1175:12: error: 'std::make_unsigned_t' has not been declared
 1175 | using std::make_unsigned_t;
      |            ^~~~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1177:12: error: 'std::remove_extent_t' has not been declared
 1177 | using std::remove_extent_t;
      |            ^~~~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1179:12: error: 'std::remove_all_extents_t' has not been declared
 1179 | using std::remove_all_extents_t;
      |            ^~~~~~~~~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1181:12: error: 'std::remove_pointer_t' has not been declared
 1181 | using std::remove_pointer_t;
      |            ^~~~~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1183:12: error: 'std::add_pointer_t' has not been declared
 1183 | using std::add_pointer_t;
      |            ^~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1185:12: error: 'std::aligned_storage_t' has not been declared
 1185 | using std::aligned_storage_t;
      |            ^~~~~~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1187:12: error: 'std::decay_t' has not been declared
 1187 | using std::decay_t;
      |            ^~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1189:12: error: 'std::common_type_t' has not been declared
 1189 | using std::common_type_t;
      |            ^~~~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1191:12: error: 'std::result_of_t' has not been declared
 1191 | using std::result_of_t;
      |            ^~~~~~~~~~~
./mbed-os/platform/cxxsupport/mstd_type_traits:1195:42: error: 'remove_cv_t' is not a member of 'std'; did you mean 'remove_cv'?
 1195 | struct remove_cvref : type_identity<std::remove_cv_t<std::remove_reference_t<T>>> { };
      |                                          ^~~~~~~~~~~
      |                                          remove_cv
./mbed-os/platform/cxxsupport/mstd_type_traits:1195:42: error: 'remove_cv_t' is not a member of 'std'; did you mean 'remove_cv'?
 1195 | struct remove_cvref : type_identity<std::remove_cv_t<std::remove_reference_t<T>>> { };
      |                                          ^~~~~~~~~~~
      |                                          remove_cv
./mbed-os/platform/cxxsupport/mstd_type_traits:1195:59: error: 'remove_reference_t' is not a member of 'std'; did you mean 'remove_reference'?
 1195 | struct remove_cvref : type_identity<std::remove_cv_t<std::remove_reference_t<T>>> { };
      |                                                           ^~~~~~~~~~~~~~~~~~
      |                                                           remove_reference
./mbed-os/platform/cxxsupport/mstd_type_traits:1195:78: error: template argument 1 is invalid
 1195 | struct remove_cvref : type_identity<std::remove_cv_t<std::remove_reference_t<T>>> { };
      |                                                                              ^
./mbed-os/platform/cxxsupport/mstd_type_traits:1195:79: error: expected '{' before '>' token
 1195 | struct remove_cvref : type_identity<std::remove_cv_t<std::remove_reference_t<T>>> { };
      |                                                                               ^~
In file included from ./mbed-os/platform/mbed_atomic.h:883,
                 from ./mbed-os/platform/SingletonPtr.h:24,
                 from ./mbed-os/drivers/AnalogIn.h:25,
                 from ./mbed-os/drivers/source/AnalogIn.cpp:18:
./mbed-os/platform/cxxsupport/mstd_type_traits:1240:10: error: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
 1240 |  -> std::enable_if_t<std::is_function<F>::value &&
      |          ^~~~~~~~~~~
      |          enable_if
./mbed-os/platform/cxxsupport/mstd_type_traits:1240:21: error: expected initializer before '<' token
 1240 |  -> std::enable_if_t<std::is_function<F>::value &&
      |                     ^
./mbed-os/platform/cxxsupport/mstd_type_traits:1250:10: error: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
 1250 |  -> std::enable_if_t<std::is_function<F>::value &&
      |          ^~~~~~~~~~~
      |          enable_if
./mbed-os/platform/cxxsupport/mstd_type_traits:1250:21: error: expected initializer before '<' token
 1250 |  -> std::enable_if_t<std::is_function<F>::value &&
      |                     ^
./mbed-os/platform/cxxsupport/mstd_type_traits:1260:10: error: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
 1260 |  -> std::enable_if_t<std::is_function<F>::value &&
      |          ^~~~~~~~~~~
      |          enable_if
./mbed-os/platform/cxxsupport/mstd_type_traits:1260:21: error: expected initializer before '<' token
 1260 |  -> std::enable_if_t<std::is_function<F>::value &&
      |                     ^
./mbed-os/platform/cxxsupport/mstd_type_traits:1271:10: error: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
 1271 |  -> std::enable_if_t<!std::is_function<F>::value &&
      |          ^~~~~~~~~~~
      |          enable_if
./mbed-os/platform/cxxsupport/mstd_type_traits:1271:21: error: expected initializer before '<' token
 1271 |  -> std::enable_if_t<!std::is_function<F>::value &&
      |                     ^
./mbed-os/platform/cxxsupport/mstd_type_traits:1281:10: error: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
 1281 |  -> std::enable_if_t<!std::is_function<F>::value &&
      |          ^~~~~~~~~~~
      |          enable_if
./mbed-os/platform/cxxsupport/mstd_type_traits:1281:21: error: expected initializer before '<' token
 1281 |  -> std::enable_if_t<!std::is_function<F>::value &&
      |                     ^
./mbed-os/platform/cxxsupport/mstd_type_traits:1291:10: error: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
 1291 |  -> std::enable_if_t<!std::is_function<F>::value &&
      |          ^~~~~~~~~~~
      |          enable_if
./mbed-os/platform/cxxsupport/mstd_type_traits:1291:21: error: expected initializer before '<' token
 1291 |  -> std::enable_if_t<!std::is_function<F>::value &&
      |                     ^
./mbed-os/platform/cxxsupport/mstd_type_traits:1302:10: error: 'enable_if_t' in namespace 'std' does not name a template type; did you mean 'enable_if'?
 1302 |  -> std::enable_if_t<!std::is_member_pointer<std::decay_t<F>>::value ||
      |          ^~~~~~~~~~~
      |          enable_if
./mbed-os/platform/cxxsupport/mstd_type_traits:1302:21: error: expected initializer before '<' token
 1302 |  -> std::enable_if_t<!std::is_member_pointer<std::decay_t<F>>::value ||
      |                     ^

[mbed] ERROR: "/usr/bin/python3" returned error.
       Code: 1
       Path: "/home/carlo/Work/Demos/Speech/3rd/tensorflow/tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed"
       Command: "/usr/bin/python3 -u /home/carlo/Work/Demos/Speech/3rd/tensorflow/tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed/mbed-os/tools/make.py -t GCC_ARM -m K66F --profile release --source . --build ./BUILD/K66F/GCC_ARM-RELEASE"
       Tip: You could retry the last command with "-v" flag for verbose output
** Further notes: **
I did install the required python packages found in mbed-os/requirements.txt so I'm not sure what the actual problem is. I've tried with both Python2.7 and Python3.8. Both pip and pip3.

--- a/tensorflow/lite/micro/examples/micro_speech/README.md
+++ b/tensorflow/lite/micro/examples/micro_speech/README.md
@@ -492,43 +492,29 @@ followed by the `K` key, then hit the `Y` key.
 
 ## Deploy to NXP FRDM K66F
 
-The following instructions will help you build and deploy the example to the
-[NXP FRDM K66F](https://www.nxp.com/design/development-boards/freedom-development-boards/mcu-boards/freedom-development-platform-for-kinetis-k66-k65-and-k26-mcus:FRDM-K66F)
-using [ARM Mbed](https://github.com/ARMmbed/mbed-cli).
+The following instructions will help you build and deploy the example to the [NXP FRDM K66F](https://www.nxp.com/design/development-boards/freedom-development-boards/mcu-boards/freedom-development-platform-for-kinetis-k66-k65-and-k26-mcus:FRDM-K66F) using [ARM Mbed](https://github.com/ARMmbed/mbed-cli).
 
-1.  Download
-    [the TensorFlow source code](https://github.com/tensorflow/tensorflow).
-2.  Follow instructions from
-    [mbed website](https://os.mbed.com/docs/mbed-os/v5.13/tools/installation-and-setup.html)
-    to setup and install mbed CLI.
+1.  Download [the TensorFlow source code](https://github.com/tensorflow/tensorflow).
+2.  Follow instructions from [mbed website](https://os.mbed.com/docs/mbed-os/v5.13/tools/installation-and-setup.html) to setup and install mbed CLI.
 3.  Compile TensorFlow with the following command to generate mbed project:
 
     ```
     make -f tensorflow/lite/micro/tools/make/Makefile TARGET=mbed TAGS="nxp_k66f" generate_micro_speech_mbed_project
     ```
 
-4.  Go to the location of the generated project. The generated project is
-    usually in
+4.  Change into the following directory that has been generated:
     `tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed`
 
-5.  Create a mbed project using the generated files: `mbed new .`
+5.  Create an Mbed project using the generated files, run ensuring your environment is using Python 2.7: 
+     `mbed config root .`
 
-6.  Change the project setting to use C++ 11 rather than C++ 14 using:
+6. Next, tell Mbed to download the dependencies and prepare to build:
+    ` mbed deploy`
 
-    ```
-    python -c 'import fileinput, glob;
-    for filename in glob.glob("mbed-os/tools/profiles/*.json"):
-      for line in fileinput.input(filename, inplace=True):
-        print line.replace("\"-std=gnu++14\"","\"-std=c++11\", \"-fpermissive\"")'
-    ```
+7. Finally, we can run the following command to compile the code:
+   `mbed compile -m K66F -t GCC_ARM` 
 
-7.  To compile project, use the following command:
-
-    ```
-    mbed compile --target K66F --toolchain GCC_ARM --profile release
-    ```
-
-8.  For some mbed compilers, you may get compile error in mbed_rtc_time.cpp. Go
+8.  For some Mbed compilers (such as GCC), you may get compile error in mbed_rtc_time.cpp. Go
     to `mbed-os/platform/mbed_rtc_time.h` and comment line 32 and line 37:
 
     ```
@@ -539,12 +525,7 @@ using [ARM Mbed](https://github.com/ARMmbed/mbed-cli).
     };
     //#endif
     ```
-
-9.  Look at helpful resources from NXP website such as
-    [NXP FRDM-K66F User guide](https://www.nxp.com/docs/en/user-guide/FRDMK66FUG.pdf)
-    and
-    [NXP FRDM-K66F Getting Started](https://www.nxp.com/document/guide/get-started-with-the-frdm-k66f:NGS-FRDM-K66F)
-    to understand information about the board.
+9. If your system does not recognize the board with the `mbed detect` command. Follow the instructions for setting up [DAPLink](https://armmbed.github.io/DAPLink/?board=FRDM-K66F) for the [K66F](https://os.mbed.com/platforms/FRDM-K66F/).
 
 10. Connect the USB cable to the micro USB port. When the Ethernet port is
     facing towards you, the micro USB port is left of the Ethernet port.
@@ -552,7 +533,7 @@ using [ARM Mbed](https://github.com/ARMmbed/mbed-cli).
 11. To compile and flash in a single step, add the `--flash` option:
 
     ```
-    mbed compile --target K66F --toolchain GCC_ARM --profile release --flash
+    mbed compile -m K66F -t GCC_ARM --flash
     ```
 
 12. Disconnect USB cable from the device to power down the device and connect

